### PR TITLE
Bug 1202974 - [Settings] Convert call settings switches to use web components

### PR DESCRIPTION
**Pull request by KevinGrandon**
https://bugzilla.mozilla.org/show_bug.cgi?id=1202974

Verified the following visuals:
- Voice privacy mode working.
- Call waiting switch - normal state is working.
- Call waiting switch - error state is working.
- Call waiting dialog switch working.

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=3cb13f398e7d4818555bd90f46b3a2f2067b4d81

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=48bdcf17b54292136aef63a42a8ae910388d170e

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=1ba8c72fdfa010df2897a5263152c978c9a867dc

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=a2b456c92bbf6323b2ada80a0000d8f8c8751ddb

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=7948a772b925d47b55ad099d5cd96e0e341039f2

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=9e858e86d680321d16714e9ccd84138898895ffa

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=eea49d67d22f97aa381846a1231e5bb4232edad7

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=ed60370f1cde2721fbb2a66aba7dcc76876df25f

**Comment by gasolin** [comment]
please rebase

File: apps/settings/elements/call.html [hunk 1]
```diff
@@ -14,19 +14,17 @@ <h1 data-l10n-id="callSettings-header"></h1>
...
         <li id="menuItem-callWaiting" aria-disabled="true" data-state="off" hidden>
-          <label class="pack-switch checkbox-label">
-            <input type="checkbox" data-ignore>
-            <span data-l10n-id="callWaiting"></span>
-          </label>
-          <label class="alert-label" hidden>
-            <span></span>
-          </label>
+          <gaia-switch data-ignore>
+            <label>
```
Comment (gasolin) [inline apps/settings/elements/call.html]: <label data-l10n-id="callWaiting"></label>

File: apps/settings/elements/call.html [hunk 2]
```diff
@@ -14,19 +14,17 @@ <h1 data-l10n-id="callSettings-header"></h1>
...
+              <span data-l10n-id="callWaiting"></span>
+            </label>
+          </gaia-switch>
+          <span class="alert" hidden></span>
```
Comment (gasolin) [inline apps/settings/elements/call.html]: `alert` class looks pretty error prone, please rename it

File: apps/settings/js/panels/call/panel.js [hunk 3]
```diff
@@ -29,13 +29,13 @@ define(function(require) {
           fdnDesc: $('#fdnSettings-desc'),
           alertPanel: $('.cw-alert'),
           alertPanelSetButton: $('.cw-alert .cw-alert-set'),
-          alertPanelConfirmInput: $('.cw-alert .cw-alert-checkbox-label input'),
+          alertPanelConfirmInput: $('.cw-alert .cw-alert-checkbox'),
           alertPanelCancelButton: $('.cw-alert .cw-alert-cancel'),
-          alertLabel: $('#menuItem-callWaiting .alert-label'),
+          alertLabel: $('#menuItem-callWaiting .alert'),
```
Comment (gasolin) [inline apps/settings/js/panels/call/panel.js]: please rename the class

File: apps/settings/style/icons.css [hunk 4]
```diff
@@ -118,14 +118,15 @@ aside.connecting {
...
 
-#menuItem-callWaiting .alert-label > span {
+#menuItem-callWaiting span.alert::after {
+  content: '';
   position: absolute;
-  right: 1.5rem;
-  width: 3rem;
+  z-index: 10;
+  top: 1.2rem;
+  right: 0;
```
Comment (gasolin) [inline apps/settings/style/icons.css]: The RTL patch is landed so there will be some changes here

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=056a288c8701870073a3c3047874080b87d275a2

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=affdff1382d5837c344b46b93fdacc4952f407fb

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=fa5894a00a0f6ede93989c9b0ea55ef46d697f9e

File: apps/settings/style/settings.css [hunk 5]
```diff
@@ -462,38 +462,31 @@ section ul.operate-certificate {
...
+#menuItem-callWaiting[data-state="on"] span.cw-alert-icon,
+#menuItem-callWaiting[data-state="off"] span.cw-alert-icon {
   display: none;
 }
 
-#menuItem-callWaiting[data-state="unknown"] .checkbox-label input ~ span:after {
-  display: none;
+#menuItem-callWaiting[data-state="unknown"] label {
+  pointer-events: none;
+  opacity: 0.6;
```
Comment (gasolin) [inline apps/settings/style/settings.css]: Previously UI designer prefer set opacity as 0.3 for gaia-switch and so on, maybe we could follow it if it does not look very strange.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=d1c23817c5dba8af018cd0c1bcb8aa3d50c16489

**Pull request merged**